analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SymbolicValueAssertions.cs

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+using SonarAnalyzer.SymbolicExecution;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+
+namespace SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution
+{
+    public class SymbolicValueAssertions : ReferenceTypeAssertions<SymbolicValue, SymbolicValueAssertions>
+    {
+        protected override string Identifier => "symbolicValue";
+
+        public SymbolicValueAssertions(SymbolicValue subject) : base(subject) { }
+
+        public AndWhichConstraint<SymbolicValueAssertions, SymbolicConstraint> HaveConstraint(SymbolicConstraint constraint, string because = "", params object[] becauseArgs)
```

Comment:
Every method in the invocation chain between `Should()` and `FailWith` should be decorated with `[CustomAssertion]`. That will tell us what exactly failed.

Imagine something like this - we currently don't know if it failed for "First", "Second" or "Third"
```
validator.Something("First").Should().BlaBla();
validator.Something("Second").Should().BlaBla();
validator.Something("Third").Should().BlaBla();
```

Reply:
I don't think this is needed here:

> Without the [CustomAssertion] attribute, Fluent Assertions would find the line that calls Should().BeTrue() and treat the customer variable as the subject-under-test (SUT). But by applying this attribute, it will ignore this invocation and instead find the SUT by looking for a call to Should().BeActive() and use the myClient variable instead.

From https://github.com/fluentassertions/fluentassertions/blob/master/docs/_pages/introduction.md#subject-identification

I don't call "Should().SomeThing()" in this method. But I can add the attribute, just to make sure if you like.